Cache httplib (#4103)
mlpack/mlpack: HISTORY.md

_????-??-??_

>  * Support caching downloaded dataset using manifest file (#4103).

Reviewer: ```suggestion
 * Add support for caching downloaded datasets to `Load()` (#4103).
```
Just a little clearer, what do you think?  I wanted to at least add `Load()` somewhere so that the reader knows what function it applies to.  It might also be worth a bullet point to indicate that we added `DownloadFile()`, up to you.

Author: I think it is fine as you proposed the cache includes the `DownloadFile()` function
